--- a/src/muse/mplugins/CMakeLists.txt.orig	2026-07-23 09:14:36.869378140 +0300
+++ b/src/muse/mplugins/CMakeLists.txt	2026-07-23 09:14:52.445615287 +0300
@@ -32,14 +32,6 @@
             ${mplugins_experimental_uis}
             )
 endif ( ENABLE_EXPERIMENTAL )
-QT5_WRAP_CPP ( mplugins_mocs
-      midifilterimpl.h
-      miditransform.h
-      midiitransform.h
-      mittranspose.h
-      mrconfig.h
-      ${experimental_hdrs}
-      )
 
 ##
 ## UI files
@@ -53,6 +45,15 @@
       )
 QT5_WRAP_UI ( mplugins_uis ${mplugins_ui_files} )
 
+QT5_WRAP_CPP ( mplugins_mocs
+      midifilterimpl.h
+      miditransform.h
+      midiitransform.h
+      mittranspose.h
+      mrconfig.h
+      ${experimental_hdrs}
+      )
+
 ##
 ## List of source files to compile
 ##
